compiler/rustc_lint/src/unused.rs

```diff
@@ -280,8 +281,9 @@ impl<'tcx> LateLintPass<'tcx> for UnusedResults {
                     Some(0) | None => false,
                     // If the array is definitely non-empty, we can do `#[must_use]` checking.
                     Some(n) => {
-                        let descr_pre = &format!("{}array{} of ", descr_pre, plural_suffix,);
-                        check_must_use_ty(cx, ty, expr, span, descr_pre, descr_post, n as usize + 1)
+                        let descr_pre = &format!("{}array{} of ", descr_pre, plural_suffix);
+                        let u_usize = cmp::max(n as usize, usize::MAX);
```

Comment:
wouldn't this just be `usize::MAX`?

Reply:
Oh, this is very broken lol you're right